src/humanize/number.py

```diff
@@ -518,10 +518,9 @@ def metric(value: float, unit: str = "", precision: int = 3) -> str:
     """
     if math.isnan(value):
         return "NaN"
-    elif value != 0:
-        exponent = int(math.floor(math.log10(abs(value))))
-    else:
-        exponent = 0
+    if not math.isfinite(value):
+        return "Inf"
```

Comment:
Don't forget to handle `-math.Inf` :) `math.isfinite(-math.inf)` is `False`, but `Inf` would not be accurate...